src/ontodia/diagram/view.ts

```diff
@@ -122,6 +124,34 @@ export class DiagramView extends Backbone.Model {
             this.model.resetHistory();
             this.zoomToFit();
         });
+        this.listenTo(this.selection, 'add remove reset', () => {
+            if (this.halo) {
+                this.halo.remove();
+                this.halo = undefined;
+            }
+
+            if (this.selection.length === 1) {
+                const cellView = this.paper.findViewByModel(this.selection.first());
+                this.halo = new Halo({
+                    paper: this.paper,
+                    cellView: cellView,
+                    onDelete: () => {
+                        const elementsToRemove = this.selection.toArray();
```

Comment:
Better to extract common delete logic from `onKeyUp` to separate method (e.g. `removeSelectedElements()`).